src/Control/Exception/Safe.hs

```diff
@@ -410,3 +414,37 @@ isAsyncException = not . isSyncException
 displayException :: Exception e => e -> String
 displayException = show
 #endif
+
+-- | You need this when using 'catches'.
+data Handler m a = forall e . (Exception e) => Handler (e -> m a)
```

Comment:
Why do we need our own `Handler` instead of reusing the one from `Control.Exception`?